src/applications/deploying-addons.md

```diff
@@ -0,0 +1,102 @@
+# Deploying Add-ons
+
+The goal of `deis addons` is to give users the power to provision consumable services such as
+a postgres database, a minio bucket, or a logstash connection) to their applications. `deis addons`
+can work with both common on-premise services such as a local mysql database or a Redis server
+for a "private" or on-premise service registry, or with public SaaS applications such as MongoLab
+or Papertrail for a public service offering.
+
+`deis addons` is backed by a project called [service-catalog][]. Service-catalog brings integration
+with [service brokers][] to the Kubernetes ecosystem via the [Open Service Broker API][].
+
+Users of Workflow use `deis addons` to provision an add-on offered by [service brokers][] for
+their applications. The end-goal is to provide a way for users to consume services from brokers and
+have their applications use those services without needing detailed knowledge about how those
+services are created or managed.
+
+As an example, most applications deployed by Workflow need a data store of some kind. `deis addons`
+allows applications to consume services like databases that exist somewhere via common environment
+variables like `DATABASE_URL`.
+
+## Listing Available Add-ons
+
+A user can use `deis addons:list` to see if their application has a database provisioned and what
+plan it is.
+
+```
+$ deis addons:list | grep -i postgresql
+$ deis-postgresql:standard
+```
+
+If the application doesn’t yet have a database provisioned, a user can create a new database using
+the CLI.
+
+## Provisioning the Add-on
+
+Most service brokers offers a variety of plans, usually spread across different tiers of service:
+hobby, standard, premium, and enterprise. For a detailed breakdown on the available plans, check
+the documentation for the applicable service broker to help choose the right service tier for the
+application.
+
+For example, to provision a `standard` plan database:
+
+```
+$ deis addons:create deis-postgresql:standard --app wooden-rowboat
+Creating deis-postgresql:standard... done
+Attaching deis-postgresql:standard to wooden-rowboat... done, v5
+```
+
+Depending on the plan, some services can take some time before it becomes available for use. Use
+`deis addons:wait` to wait for the service to become available.
+
+Once the service has been attached to the application, a DATABASE_URL setting will be available in
+the application's configuration and will contain the URL used to access the newly provisioned
+service.
+
+You can choose the alias that the add-on uses on the application using the `--as` flag. This will
```

Comment:
I was thinking we could do some crazy docker-like mapping in the form of `--as DATABASE_URL=DB_URL,DATABASE_USER=DB_USER`, but again I'm concerned about making the translation overly complex. I'm sure this will come into play as things start being implemented.